### ISSUE TITLE
Kodi 19 fix please

### ISSUE THREAD

**Issue by audiowin** [chrisbevan/skin.pellucid#63]
Unfortunately, you cannot make any new start buttons because the function does not work properly (user-defined)

**Comment by EnerGeh** [comment]
Unfortunately your issue title is very general. What you mean is the fact that currently new menu entries can be created but not customized. IMHO this is due to missing dependencies and will hopefully be fixed when installed  ia Kodi Addon Repo. The author is currently waiting for the merge in the matrix repos.

**Comment by EnerGeh** [comment]
Okay, got the Plugin Form Repo and issue still exists:
When you create an new Menu entry and choose 'Change action', you get a popup with 'Choose Item For Menu' with an empty list. Hit OK and you will be prompted to install widget providers where 'ExtendedInfo Script' fails to install, this causes a loop in the popup menus.
Tested on LibreELEC 9.95.1 with Matrix on x86 Hardware.

**Comment by chrisbevan** [comment]
Nothing I can do about this i'm afraid. People will have to wait until whoever maintains script.extendedinfo updates to Py3.

**Comment by EnerGeh** [comment]
I got your point and please excuse the hazzle, but is that addon required for the customization of the menu? I didn't find any other way to customize the menu as it loops around these two dialogs? For example: I'd like to add a favorites menu entry to the homescreen. How would you suggest to do that?

**Comment by chrisbevan** [comment]
Yes. The whole menu system is managed by the skin shortcuts script. You can manually override by adding entries to skin.pellucid->shortcuts->mainmenu.DATA.xml. The process is not straightforward however and i'm afraid you will need to reverse engineer your way through without my support.

The Matrix update has obviously broken many things. I'd advise patience. Many of these addons are well used and will get updated in due course.

Sorry can't be more help. This is a hobby for me - I simply don't have the time to offer in depth technical support. I hope you understand.

**Comment by EnerGeh** [comment]
Thanks for your answer, I just needed the hint where to look for that file. Found it and will try to modify by myself.
I see that the change to Python 3 was one of the biggest challenges in the past years and fully understand the frustration of the developers and contributors to that project concerning user complaints. So please accept my honest thanks for your great work and keep on coding!

**Comment by johnsmiller62** [comment]
Great skin @chrisbevan - thanks for your work.

So I have got the menu items dialog working so that you can pick targets in Pellucid v3.0.0.0 menus.  Several Python 3 addons need to be manually installed in the following order:

1. scott967 / script.module.kodi65 - https://github.com/scott967/script.module.kodi65/tree/Matrix
2. kodi-community-addons / script.module.musicbrainz - https://github.com/kodi-community-addons/script.module.musicbrainz
3. kodi-community-addons / script.module.thetvdb - https://github.com/kodi-community-addons/script.module.thetvdb
4. kodi-community-addons / script.module.metadatautils - https://github.com/kodi-community-addons/script.module.metadatautils
5. kodi-community-addons / script.skin.helper.service - https://github.com/kodi-community-addons/script.skin.helper.service
6. scott967 / script.extendedinfo - https://github.com/scott967/script.extendedinfo/tree/Matrix

And then finally I has to edit a file in script.skinshortcuts at addons\script.skinshortcuts\resources\lib\library.py - comment out line 2146 as follows #self.getControl(7).setLabel(xbmc.getLocalizedString(222))

I'm sure the last step may break other addons if they use script.skinshortcuts but I could not get Pellucid working without it. In Pellucid it seems to be in DialogSelect.xml at line 310 for the cancel button that the problem happens but even setting id to 7 here did not resolve the problem without commenting out line 2146 in addons\script.skinshortcuts\resources\lib\library.py

**Comment by Badsheep** [comment]
Hi @johnsmiller62,

I just tried your workaround but it's doesn't seem to work on my Kodi.
(I use Kodi 19.3 on a mac os 11.6.1 hackintosh)

when I install the step 6, Kodi becomes like crazy, creates new instance of itself, and display error messages.
(if I let it do, it keeps on creating new and new Kodi applications)

I have other addons installed, Netflix and Disney+.
Like you said, this step 6 may break some toher stuffs...

[Log here.](https://paste.kodi.tv/tuxofisupu.kodi)

**Comment by chrisbevan** [comment]
Hi all,

Note that the most recent version of Pellucid on this repository has removed the user-editable menu system. There is no need for any of these workarounds.

**Comment by Badsheep** [comment]
Hi,

Yes but I DO need this user-editable menu :)

**Comment by chrisbevan** [comment]
Understood. Please do be aware that I will not be bringing it back, even when the rest of the problems with these third party plugins are fixed.

I have however kept the separate menus for music / videos / games / pictures and the homescreen. It may be easier for you to simply edit those files directly (https://github.com/chrisbevan/skin.pellucid/blob/master/1080i/Includes_menus.xml)

**Comment by Badsheep** [comment]
Ok thanks I’ll have a look at that.
Do you think I could use shorcuts in addon ? And shortcut to a python script ? That’s what I was using in my Kodi configuration.

In case it’s not possible, could you provide my a link to your last v.2.x that was using the skin shorcut ? In case of
Thanks !

**Comment by Badsheep** [comment]
Hi everybody it was an hard fight but I made it :)
If you want to use Pellucid 3.0.0 with custom actions, [follow the steps](https://github.com/chrisbevan/skin.pellucid/issues/63#issuecomment-895753526) of @johnsmiller62 except : 

- at step 5, you need another version, [see here](https://github.com/kodi-community-addons/script.skin.helper.service/issues/144)

- the line that has to be commented is not anymore in the file "library.py", it's in the file /addons/script.skinshortcuts/resources/lib/skinshorcuts/common_utils.py at line #49.

Yiha !

And thank you one more time @chrisbevan for that really nice skin, I spent time trying others and there's no comparison :)

**Comment by johnsmiller62** [comment]
Hey @Badsheep great work and thanks for building on what I did. Tried your workaround to my workaround and works great as you said.

@chrisbevan pity the user editable menus are removed in the latest version of Pellucid, but quite understand the reason.

**Comment by konrul-r10** [comment]
> Hi everybody it was an hard fight but I made it :) If you want to use Pellucid 3.0.0 with custom actions, [follow the steps](https://github.com/chrisbevan/skin.pellucid/issues/63#issuecomment-895753526) of @johnsmiller62 except :
> 
> * at step 5, you need another version, [see here](https://github.com/kodi-community-addons/script.skin.helper.service/issues/144)
> * the line that has to be commented is not anymore in the file "library.py", it's in the file /addons/script.skinshortcuts/resources/lib/skinshorcuts/common_utils.py at line 49.
> 
> Yiha !
> 
> And thank you one more time @chrisbevan for that really nice skin, I spent time trying others and there's no comparison :)

Hi, could you please explain the details, please? Many people -including me- are not experts in compiling programs or Linux OS. We will be appriciated if you could explain how to do that, thank you :)

**Comment by Badsheep** [comment]
Hello,
No need to compil, and this procedure does not depend of the OS I suppose. I use it with Mac OS.

Just install KODI, and then you just will have to install Addons from ZIP files
(see [here](https://kodi.wiki/view/Add-on_manager) if you don't know how to do that)

Then follow the steps below. For each step, you click on the link, on the opened page you click on the green "code" button, then on the menu you select "download ZIP". It's that zip that you install on kodi. Except on the link of step 5, which is directly the Zip you have to install.

0 - Install Pellucid 3.0.0 from Kodi repository.
      When installed don't switch to that skin, stay on Estuary (not sure it's necessary but that's what I did)
1 - scott967 / script.module.kodi65 - https://github.com/scott967/script.module.kodi65/tree/Matrix
2 - kodi-community-addons / script.module.musicbrainz - https://github.com/kodi-community-3- addons/script.module.musicbrainz
3 - kodi-community-addons / script.module.thetvdb - https://github.com/kodi-community-addons/script.module.thetvdb
4 - kodi-community-addons / script.module.metadatautils - https://github.com/kodi-community-addons/script.module.metadatautils
5 - Marcelveldt / repository - https://github.com/kodi-community-addons/repository.marcelveldt/raw/master/repository/repository.marcelveldt/repository.marcelveldt-1.0.3.zip
6 - script.skin.helper.service / From Marcel's repository, install script.skin.helper.service
7 - scott967 / script.extendedinfo - https://github.com/scott967/script.extendedinfo/tree/Matrix

When everything is installed, go to file /addons/script.skinshortcuts/resources/lib/skinshorcuts/common_utils.py and comment the line 49 (add a # in the beginning of this line).

Good luck !
And happy Christmas :)

**Comment by konrul-r10** [comment]
Hello @Badsheep first of all, thank you for your instructions, it has worked! 

There is only problem at here: kodi-community-addons / script.module.metadatautils - https://github.com/kodi-community-addons/script.module.metadatautils 
kodi couldn't install it. 
What is it for? What if I won't install it?

My system is Raspbian bullseye with Raspberry Pi 4 (4GB) kodi is v19

Maybe, that is why that add on not work on my kodi...

Best regards, thank you for your help and Happy Christmas!

**Comment by Badsheep** [comment]
Hello,

Sorry I'm not home, I can't check the version I use...
I'll look at that back home !
Maybe they did breaking changes... does it work without ?

**Comment by konrul-r10** [comment]
Thank you. I am also away from my device. I will try it soon. Thanks again. Best regards!
